### PR TITLE
[ci] Update CI to ensure SAI spec are generated and updated.

### PR DESCRIPTION
**Pull request by r12f**
To help the code review of SAI header updates, update the CI workflow to ensure we have up-to-date SAI spec included in the PR.

This is achieved by checking git status and see if any changes in the SAI/specs folder, after running `make sai` in the CI pipeline.

### CODE REVIEW

**Comment by r12f** [comment]
Need #594 to be checked in to make it stable.

**Comment by KrisNey-MSFT** [comment]
hi @chrispsommers - did you happen to check this one?  It had requested your review...TY, Kristina

File: dash-pipeline/Makefile [hunk 1]
```diff
@@ -237,6 +237,14 @@ sai-clean: SAI/SAI libsai-clean saithrift-server-clean
 	rm -rf SAI/SAI/inc SAI/SAI/experimental SAI/SAI/meta
 	cd SAI/SAI && git checkout -- inc experimental meta
 
+.PHONY:check-sai-spec
+check-sai-spec:
+	if git status | grep -q "SAI/specs"; then \
```
Comment (chrispsommers) [inline dash-pipeline/Makefile:242]: What causes the exit status to change from 1 (fail) to 0 (pass)? If the developer commits headers locally for review does this somehow remove the failure?

Comment (r12f) [reply dash-pipeline/Makefile:242]: Hi Chris, thanks a lot for reviewing the change! 

This change is to only ensure all the SAI API changes caused by the PR will be reflected in PR as changes to SAI specs. So this task only runs in the CI pipeline and nowhere else. And the condition to pass the check is simple, as long as we have run 'make sai-headers' locally and adds the SAI spec updates in, this check will pass.

Comment (chrispsommers) [reply dash-pipeline/Makefile:242]: ok, thanks!

**Review by chrispsommers — APPROVED** [review]

File: dash-pipeline/Makefile (hunk 1 above)
Comment (r12f) [reply dash-pipeline/Makefile:242]: Thanks Chris!

**Pull request merged**